File: problem-solving/baekjoon/3986_좋은_단어.py
# https://www.acmicpc.net/problem/3986
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    count = 0
    for i in range(n):
        inputs = list(sys.stdin.readline().rstrip())
        stack = [inputs[0]]
        for j in range(1, len(inputs)):
            if len(stack) == 0:
                stack = [inputs[j]]
            elif stack[-1] == inputs[j]:
                stack.pop()
            else:
                stack.append(inputs[j])
        if len(stack) == 0:
            count += 1
    print(count)

File: problem-solving/baekjoon/3986_좋은_단어.go
// https://www.acmicpc.net/problem/3986
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)
	var count int
	for i := 0; i < n; i++ {
		var input string
		input, _ = reader.ReadString('\n') // 공백 포함하여 입력 받기 위해 ReadString() 사용
		inputs := strings.Split(strings.ReplaceAll(input, "\n", ""), "")
		stack := []string{inputs[0]}
		for j := 1; j < len(inputs); j++ {
			if len(stack) == 0 {
				stack = []string{inputs[j]}
			} else if stack[len(stack)-1] == inputs[j] {
				stack = stack[:len(stack)-1]
			} else {
				stack = append(stack, inputs[j])
			}
		}
		if len(stack) == 0 {
			count++
		}
	}
	fmt.Fprintln(writer, count)
}
